consensus/src/common.rs

```diff
@@ -120,21 +124,23 @@ impl Chain<ProtocolState> for ProtocolStateChain {
         self.0.len()
     }
 
-    fn last_vrf_hash(&self) -> Option<String> {
-        self.top().map(|s| {
-            s.body
-                .consensus_state
-                .last_vrf_output
-                .hash()
-                .as_ref()
-                .encode_hex::<String>()
-        })
+    fn last_vrf_hash(&self) -> Option<Fp> {
+        static HASHER_POOL: OnceCell<SpinLockObjectPool<PoseidonHasherKimchi<VrfOutputTruncated>>> =
```

Comment:
Let's move this pool initialization block at the top as a function which returns `pool`. I see we initialize pool both in `state_hash()` and `last_vrf_hash()`